script.js

```diff
@@ -1,4 +1,608 @@
+  gamePhase = 2;
+  return message;
+};
+
+// Function: Game Phase 2 (Choice - Standard)
+var phase2 = function (choice) {
+  finalNumbers.push(chosenOrderNum(choice, playerRolls));
+  var message = `<b>Player ${playerTurn}</b>, you chose Dice ${choice} first.<br><br>Your number is <b>${
+    finalNumbers[playerTurn - 1]
+  }</b>`;
+  return message;
+};
+
+// Function: Game Phase 2 (Choice - Variable Number of Dice)
+var phase2VariableDice = function () {
+  var index = 0;
+  if (gameMode == "Highest Mode") {
+    var number = 6;
+    var highestNum = "";
+    while (number > 0) {
+      index = (playerTurn - 1) * (numOfDice + 1) + 1;
+      while (
+        index > (playerTurn - 1) * (numOfDice + 1) &&
+        index < playerTurn * (numOfDice + 1)
+      ) {
+        if (playerRolls[index] == number) {
+          highestNum += playerRolls[index];
+        }
+        index += 1;
+      }
+      number -= 1;
+    }
+    finalNumbers.push(Number(highestNum));
+    return `Your highest number is <b>${highestNum}</b>`;
+  } else if (gameMode == "Lowest Mode") {
+    var number = 0;
+    var lowestNum = "";
+    while (number <= 6) {
+      index = (playerTurn - 1) * (numOfDice + 1) + 1;
+      while (
+        index > (playerTurn - 1) * (numOfDice + 1) &&
+        index < playerTurn * (numOfDice + 1)
+      ) {
+        if (playerRolls[index] == number) {
+          lowestNum += playerRolls[index];
+        }
+        index += 1;
+      }
+      number += 1;
+    }
+    finalNumbers.push(Number(lowestNum));
+    return `Your lowest number is <b>${lowestNum}</b>`;
+  }
+};
+
+// Function: Game Phase 2 (Choice - Auto-Generated Combined Number)
+var phase2Auto = function () {
+  var diceRoll1 = playerRolls[(playerTurn - 1) * 3 + 1];
+  var diceRoll2 = playerRolls[(playerTurn - 1) * 3 + 2];
+  if (gameMode == "Highest Mode") {
+    if (diceRoll1 >= diceRoll2) {
+      finalNumbers.push(chosenOrderNum(1, playerRolls));
+    } else if (diceRoll2 > diceRoll1) {
+      finalNumbers.push(chosenOrderNum(2, playerRolls));
+    }
+  } else if (gameMode == "Lowest Mode") {
+    if (diceRoll1 <= diceRoll2) {
+      finalNumbers.push(chosenOrderNum(1, playerRolls));
+    } else if (diceRoll2 < diceRoll1) {
+      finalNumbers.push(chosenOrderNum(2, playerRolls));
+    }
+  }
+  var message = `Player ${playerTurn}, your number is <b>${
+    finalNumbers[playerTurn - 1]
+  }</b>`;
+  return message;
+};
+
+// Function: Game Phase 2 (Choice - Knockout)
+var phase2Knockout = function (choice) {
+  var chosenNum = 0;
+  if (choice == 1) {
+    chosenNum = Number(
+      String(playerRolls[(knockoutRound - 1) * 3 + 1]) +
+        String(playerRolls[(knockoutRound - 1) * 3 + 2])
+    );
+  } else if (choice == 2) {
+    chosenNum = Number(
+      String(playerRolls[(knockoutRound - 1) * 3 + 2]) +
+        String(playerRolls[(knockoutRound - 1) * 3 + 1])
+    );
```

Comment:
this could be refactored into a function that tooks rolls [roll1, roll2] and choice. You have the same logic above, right now it couldn't be refactored because your roll number is tangled with your knockoutRound 
separate the getting-of-the-roll-number logic and the generation of the chosenNum and you can refactor it!